[Ingest Manager] Document fleet configuration flags (#66736)
elastic/kibana: docs/settings/ingest-manager-settings.asciidoc

[role="xpack"]
[[ingest-manager-settings-kb]]

> === Ingest Manager settings in Kibana

Reviewer: ```suggestion
=== Ingest Manager settings in {kib}
```

++++
<titleabbrev>Ingest Manager settings</titleabbrev>
++++

You can configure `xpack.ingestManager` settings in your `kibana.yml`

[[general-ingest-manager-settings-kb]]
==== General Ingest Manager settings

[cols="2*<"]
|===
| `xpack.ingestManager.enabled`

>   | Set to `true` to enable {kib} Ingest Manager. 

Reviewer: ```suggestion
  | Set to `true` to enable Ingest Manager. 
```

|===

[cols="2*<"]
|===
| `xpack.ingestManager.epm.enabled`

>   | Set to `true` (default) to enable {kib} EPM. 

Reviewer: Avoid abbreviations like EPM, especially for new technology. I think we calling this simple "package manager" (see https://github.com/elastic/style-guide/issues/82#issuecomment-637005667).

Since product names are volatile, I will created some shared attributes that you can use to resolve these names.

Reviewer: Update: I've opened a PR that contains the attributes. Just need to get the PM to review. If that PR isn't merged before you're ready, I can do a follow-up PR. Just let me know. https://github.com/elastic/docs/pull/1862

Author: Perfect I will make the change or you will depending on which PR is merged before

>   | Set to `true` (default) to enable {kib} EPM. 

Reviewer: ```suggestion
  | Set to `true` (default) to enable the package manager. 
```

Reviewer: Actually...wait on this one. Still sorting out the correct name.

|===

[cols="2*<"]
|===
| `xpack.ingestManager.fleet.enabled`

>   | Set to `true` (default) to enable {kib} Fleet. 

Reviewer: ```suggestion
  | Set to `true` (default) to enable Fleet. 
```

|===

[[ingest-manager-data-visualizer-settings]]

==== Fleet settings

[cols="2*<"]
|===
| `xpack.ingestManager.fleet.kibana.host`

>   | The hostname used by Elastic Agent for accessing Kibana

Reviewer: ```suggestion
  | The hostname used by Elastic Agent for accessing {kib}.
```

|===

[cols="2*<"]
|===
| `xpack.ingestManager.fleet.elasticsearch.host`

>   | The hostname used by Elastic Agent for accessing ElasticSearch

Reviewer: ```suggestion
  | The hostname used by Elastic Agent for accessing {es}.
```

|===

[cols="2*<"]
|===
| `xpack.ingestManager.fleet.tlsCheckDisabled`
  | Set to `true` to allow to run Fleet on a Kibana instance without TLS enabled.
|===

[NOTE]
====

> In Elastic Cloud, Fleet flags are already configured.

Reviewer: ```suggestion
In {ecloud}, Fleet flags are already configured.
```

[role="xpack"]
[[ingest-manager-settings-kb]]

> === Ingest Manager settings in {kib}

Reviewer: ```suggestion
=== {ingest-manager} settings in {kib}
```

++++

> <titleabbrev>Ingest Manager settings</titleabbrev>

Reviewer: ```suggestion
<titleabbrev>{ingest-manager} settings</titleabbrev>
```

++++

You can configure `xpack.ingestManager` settings in your `kibana.yml`. 

> By default Ingest Manager is not enabled you need to enable it. Also to you use Fleet, you need to configure {kib} and {es} hosts.

Reviewer: ```suggestion
By default, {ingest-manager} is not enabled. You need to enable it. To use {fleet}, you also need to configure {kib} and {es} hosts.
```

[[general-ingest-manager-settings-kb]]

> ==== General Ingest Manager settings

Reviewer: ```suggestion
==== General {ingest-manager} settings
```

[cols="2*<"]
|===
| `xpack.ingestManager.enabled` {ess-icon}

>   | Set to `true` to enable Ingest Manager. 

Reviewer: ```suggestion
  | Set to `true` to enable {ingest-manager}. 
```

|===

[cols="2*<"]
|===
| `xpack.ingestManager.epm.enabled` {ess-icon}

>   | Set to `true` (default) to enable the package manager. 

Reviewer: ```suggestion
  | Set to `true` (default) to enable {package-manager}. 
```

|===

[cols="2*<"]
|===
| `xpack.ingestManager.fleet.enabled` {ess-icon}

>   | Set to `true` (default) to enable Fleet. 

Reviewer: ```suggestion
  | Set to `true` (default) to enable {fleet}. 
```

|===

[[ingest-manager-data-visualizer-settings]]

> ==== EPM settings

Reviewer: ```suggestion
==== {package-manager} settings
```

Reviewer: I am wondering if I should be using {package-registry} here instead of {package-manager}?

[cols="2*<"]
|===
| `xpack.ingestManager.epm.registryUrl`

>   | The address use to reach EPM registry

Reviewer: ```suggestion
  | The address to use to reach {package-manager} registry.
```

|===

> ==== Fleet settings

Reviewer: ```suggestion
==== {fleet} settings
```

[cols="2*<"]
|===
| `xpack.ingestManager.fleet.kibana.host`

>   | The hostname used by Elastic Agent for accessing {kib}.

Reviewer: ```suggestion
  | The hostname used by {agent} for accessing {kib}.
```

|===

[cols="2*<"]
|===
| `xpack.ingestManager.fleet.elasticsearch.host`

>   | The hostname used by Elastic Agent for accessing {es}.

Reviewer: ```suggestion
  | The hostname used by {agent} for accessing {es}.
```

|===

[cols="2*<"]
|===
| `xpack.ingestManager.fleet.tlsCheckDisabled`

>   | Set to `true` to allow Fleet to run on a {kib} instance without TLS enabled.

Reviewer: ```suggestion
  | Set to `true` to allow {fleet} to run on a {kib} instance without TLS enabled.
```

|===

[NOTE]
====

> In {ecloud}, Fleet flags are already configured.

Reviewer: ```suggestion
In {ecloud}, {fleet} flags are already configured.
```
